angularSTEP/src/app/discover-display/discover-display.component.ts

```diff
@@ -64,67 +65,77 @@ export class DiscoverDisplayComponent implements OnInit {
   }
 
   getRecipes() {
-    switch (this.sortType) {
-      case 'Time Created': {
-        if (this.direction) {
-          this.recipeCollection.ref
-            .orderBy('timestamp', 'asc')
-            .withConverter(new RecipeConverter().recipeConverter)
-            .get()
-            .then(recipes => {
-              this.recipes = recipes.docs;
-            });
-        } else {
-          this.recipeCollection.ref
-            .orderBy('timestamp', 'desc')
-            .withConverter(new RecipeConverter().recipeConverter)
-            .get()
-            .then(recipes => {
-              this.recipes = recipes.docs;
-            });
+    if(this.query){
+      this.recipeCollection.ref
+              .where('tags', 'array-contains', this.query)
+              .withConverter(new RecipeConverter().recipeConverter)
+              .get()
+              .then(recipes => {
+                this.recipes = recipes.docs;
+              });
+    } else {
+      switch (this.sortType) {
+        case 'Time Created': {
+          if (this.direction) {
+            this.recipeCollection.ref
+              .orderBy('timestamp', 'asc')
+              .withConverter(new RecipeConverter().recipeConverter)
+              .get()
+              .then(recipes => {
+                this.recipes = recipes.docs;
+              });
+          } else {
+            this.recipeCollection.ref
+              .orderBy('timestamp', 'desc')
+              .withConverter(new RecipeConverter().recipeConverter)
+              .get()
+              .then(recipes => {
+                this.recipes = recipes.docs;
+              });
+          }
+          break;
         }
-        break;
-      }
-      case 'Number of Ingredients': {
```

Comment:
Is sort by 'Number of Ingredients' being removed on purpose? Please clarify in the change description! AFAICT this looks like an error where the whole sorting option has been accidentally deleted

Reply:
Oh yes, thank you. I always write about what I added but not what I deleted! Just updated the change description.